File: src/test/java/com/travel/domain/travel/service/TravelServiceTest.java
package com.travel.domain.travel.service;

import com.travel.domain.travel.domain.JavaTravel;
import com.travel.domain.travel.domain.TravelRepository;
import com.travel.domain.travel.dto.*;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
public class TravelServiceTest {

    @Autowired
    TravelService travelService;

    @Autowired
    TravelRepository travelRepository;

    @AfterEach
    public void clean() {
        travelRepository.deleteAll();
    }

    @DisplayName("[단위테스트] 여행 등록")
    @Test
    public void save() {
        //given
        travelService.save(TravelSaveDto.builder()
                .name("대구여행 렛츠 고")
                .userId("gogogo")
                .travelStartDate("20221229")
                .travelEndDate("20230101")
                .build());

        //when
        List<JavaTravel> list = travelRepository.findAll();

        //then
        assertThat(list.get(0).getName()).isEqualTo("대구여행 렛츠 고");
    }

    @DisplayName("[단위테스트] 여행 수정")
    @Test
    public void update() throws Exception {
        //given
        Long saveCity = travelService.save(TravelSaveDto.builder()
                .name("대구여행 렛츠 고")
                .userId("gogogo")
                .travelStartDate("20221229")
                .travelEndDate("20230101")
                .build());

        //when
        travelService.update(1L, TravelUpdateDto.builder()
                .name("전주여행 렛츠 고")
                .travelStartDate("20221230")
                .travelEndDate("20230101")
                .build());

        //then
        List<JavaTravel> travel = travelRepository.findAll();
        assertThat(travel.get(0).getName()).isEqualTo("전주여행 렛츠 고");
        assertThat(travel.get(0).getTravelStartDate()).isEqualTo("20221230");
        assertThat(travel.get(0).getTravelEndDate()).isEqualTo("20230101");
    }

    @DisplayName("[단위테스트] 여행 삭제 ")
    @Test
    public void delete() {
        //given
        Long saveCity = travelService.save(TravelSaveDto.builder()
                .name("대구여행 렛츠 고")
                .userId("gogogo")
                .travelStartDate("20221229")
                .travelEndDate("20230101")
                .build());

        //when
        travelService.delete(saveCity);

        //then
        List<JavaTravel> travelList = travelRepository.findAll();
        assertThat(travelList.size()).isEqualTo(0);
    }

    @DisplayName("[단위테스트] 여행 삭제 ")
    @Test
    public void get() {
        TravelResponseDto responseDto = travelService.get(TravelRequestDto.builder().id(10L).searchType(TravelSearchType.PLAN).build());

        assertThat(responseDto.getId()).isEqualTo(10L);
    }

}

File: src/test/kotlin/com/travel/domain/travel/TravelServiceTest.kt
package com.travel.domain.travel.service

import com.travel.domain.travel.domain.Travel
import com.travel.domain.travel.domain.TravelRepository
import com.travel.domain.travel.dto.*
import org.assertj.core.api.Assertions
import org.assertj.core.api.AssertionsForInterfaceTypes.assertThat
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
class TravelServiceTest @Autowired constructor(
        private var travelService: TravelService,
        private var travelRepository: TravelRepository,
) {
    @AfterEach
    fun clean() {
        travelRepository.deleteAll()
    }

    @DisplayName("[단위테스트] 여행 등록")
    @Test
    fun save() {
        //given
        travelService.save(TravelSaveDto(
                name = "대구여행 렛츠 고",
                userId = "gogogo",
                travelStartDate = "20221229",
                travelEndDate = "20230101"
        ))

        //when
        val list: List<Travel> = travelRepository!!.findAll()

        //then
        assertThat(list[0].name).isEqualTo("대구여행 렛츠 고")
    }

    @DisplayName("[단위테스트] 여행 수정")
    @Test
    @Throws(Exception::class)
    fun update() {
        //given
        travelService.save(TravelSaveDto(
                name = "대구여행 렛츠 고",
                userId = "gogogo",
                travelStartDate = "20221229",
                travelEndDate = "20230101"
        ))

        var saveKey: Long = travelRepository.findAll()[0].id!!

        //when
        travelService.update(saveKey, TravelUpdateDto(
                name = "대구여행 렛츠 고",
                travelStartDate = "20221229",
                travelEndDate = "20230101"))

        //then
        val travel: List<Travel> = travelRepository.findAll()
        assertThat(travel[0].name).isEqualTo("전주여행 렛츠 고")
        assertThat(travel[0].travelStartDate).isEqualTo("20221230")
        assertThat(travel[0].travelEndDate).isEqualTo("20230101")
    }

    @DisplayName("[단위테스트] 여행 삭제 ")
    @Test
    fun delete() {
        //given
        travelService.save(TravelSaveDto(
                name = "대구여행 렛츠 고",
                userId = "gogogo",
                travelStartDate = "20221229",
                travelEndDate = "20230101"
        ))

        //when
        travelService.delete(travelRepository.findAll()[0].id!!)

        //then
        val travelList: List<Travel> = travelRepository.findAll()
        Assertions.assertThat(travelList.size).isEqualTo(0)
    }

//    @DisplayName("[단위테스트] 여행 삭제 ")
//    @Test
//    fun get() {
//        val responseDto: TravelResponseDto = travelService.get(TravelRequestDto(
//                id = 10L,
//                searchType = TravelSearchType.PLAN))
//        assertThat(responseDto.id).isEqualTo(10L)
//    }
}